src/rust/dbv2/migrations/2024-01-17-184459_coingecko/up.sql

```diff
@@ -0,0 +1,192 @@
+        ELSE size_and_price_to_quote_indivisible_subunits($1,remaining_size,mid_price)
+    END)
+    AS liquidity_in_quote
+    FROM api.orders, mid_price
+    WHERE order_status = 'open'
+    AND market_id = $1
+    AND price::numeric BETWEEN mid_price::numeric * (1::numeric - $2::numeric/10000::numeric) AND mid_price::numeric * (1::numeric + $2::numeric/10000::numeric);
+$$ LANGUAGE SQL;
+
+
+CREATE FUNCTION api.get_market_24h_low (market_id numeric) RETURNS NUMERIC AS $$
+    SELECT MIN(price)
+    FROM fill_events
+    WHERE "time" > CURRENT_TIMESTAMP - interval '1 day'
+    AND fill_events.market_id = $1;
+$$ LANGUAGE SQL;
+
+
+CREATE FUNCTION api.get_market_24h_high (market_id numeric) RETURNS NUMERIC AS $$
+    SELECT MAX(price)
+    FROM fill_events
+    WHERE "time" > CURRENT_TIMESTAMP - interval '1 day'
+    AND fill_events.market_id = $1;
+$$ LANGUAGE SQL;
+
+
+CREATE FUNCTION api.get_market_ask (market_id numeric) RETURNS NUMERIC AS $$
+    SELECT MIN(price)
+    FROM api.orders
+    WHERE orders.market_id = $1
+    AND order_status = 'open'
+    AND direction = 'ask';
+$$ LANGUAGE SQL;
+
+
+CREATE FUNCTION api.get_market_bid (market_id numeric) RETURNS NUMERIC AS $$
+    SELECT MAX(price)
+    FROM api.orders
+    WHERE orders.market_id = $1
+    AND order_status = 'open'
+    AND direction = 'bid';
+$$ LANGUAGE SQL;
+
+
+CREATE FUNCTION api.get_market_last_price (market_id numeric) RETURNS NUMERIC AS $$
+    SELECT price
+    FROM fill_events
+    ORDER BY txn_version DESC, event_idx DESC
+    LIMIT 1;
+$$ LANGUAGE SQL;
+
+
+-- Create the views
+CREATE VIEW api.tickers AS
+SELECT
+    market_id,
+    base_account_address || '::' || base_module_name || '::' || base_struct_name AS base_currency,
+    quote_account_address || '::' || quote_module_name || '::' || quote_struct_name AS quote_currency,
+    base_volume_24h AS base_volume,
+    quote_volume_24h AS quote_volume,
+    integer_price_to_quote_nominal(market_id, api.get_market_last_price(market_id)) AS last_price,
+    integer_price_to_quote_nominal(market_id, api.get_market_ask(market_id)) AS ask,
+    integer_price_to_quote_nominal(market_id, api.get_market_bid(market_id)) AS bid,
+    integer_price_to_quote_nominal(market_id, api.get_market_24h_high(market_id)) AS high,
+    integer_price_to_quote_nominal(market_id, api.get_market_24h_low(market_id)) AS low,
+    api.get_market_liquidity(market_id,200) / get_quote_volume_divisor_for_market(market_id) AS liquidity_in_quote
+FROM
+    api.markets;
+
+
+CREATE FUNCTION api.orderbook (market_id numeric, depth numeric) RETURNS TABLE(market_id numeric(20,0), txn_version numeric(20,0), bids numeric[], asks numeric[]) AS $$
+    WITH mid_price AS (
+        SELECT
+          (MIN(price) FILTER (WHERE direction = 'ask') + MAX(price) FILTER (WHERE direction = 'bid')) / 2 AS mid_price
+        FROM api.orders
+        WHERE orders.market_id = $1 AND order_status = 'open'
+    ),
```

Comment:
This was already included as a CTE above, how about breaking this out to be its own function? (And adding a down migration too)